### DISCUSSION TITLE
安装v2ray 后，指定的端口还是TCP国内外均不可用 #980

### DISCUSSION THREAD

**Discussion by colintheunique** [v2fly/v2ray-core#2237] [Q&A]
## 你正在使用哪个版本的 V2Ray？
5.1.0

## 你的使用场景是什么？

正常翻墙

## 你看到的异常现象是什么？
安装完后端口还是TCP不可用，国内外均不可用
安装过程中，出现fastmirror链接超时，启用cache 镜像：
Could not retrieve mirrorlist http://mirrors.elrepo.org/mirrors-elrepo.el7 error was
12: Timeout on http://mirrors.elrepo.org/mirrors-elrepo.el7: (28, 'Connection timed out after 30001 milliseconds')

    base: mirror.xeonbd.com
    elrepo: mirrors.coreix.net
    epel: repo.extreme-ix.org
    extras: mirror.xeonbd.com
    updates: mirror.aktkn.sg
    No packages marked for update

Package lrzsz-0.12.20-36.el7.x86_64 already installed and latest version
Package git-1.8.3.1-23.el7_8.x86_64 already installed and latest version
Package zip-3.0-11.el7.x86_64 already installed and latest version
Package unzip-6.0-24.el7_9.x86_64 already installed and latest version
Package curl-7.29.0-59.el7_9.1.x86_64 already installed and latest version
Package wget-1.14-18.el7_6.1.x86_64 already installed and latest version
Package qrencode-3.4.1-3.el7.x86_64 already installed and latest version
Package libcap-2.22-11.el7.x86_64 already installed and latest version
Package jq-1.6-2.el7.x86_64 already installed and latest version
Nothing to do
Cloning into '/tmp/233boy/v2ray'...
remote: Enumerating objects: 89, done.
remote: Counting objects: 100% (89/89), done.
remote: Compressing objects: 100% (59/59), done.
remote: Total 89 (delta 29), reused 68 (delta 27), pack-reused 0
Unpacking objects: 100% (89/89), done.

## 你期待看到的正常表现是怎样的？
可翻墙

## 请附上你的配置

**服务端配置：**

```javascript
// 在这里附上服务器端配置文件

```

**客户端配置：**

```javascript
// 在这里附上客户端配置

```

## 请附上出错时软件输出的错误日志

**服务器端错误日志：**

```javascript
// 在这里附上服务器端日志

```

**客户端错误日志：**

```javascript
// 在这里附上客户端日志

```

## 请附上访问日志

```javascript
// 在这里附上服务器端日志

```

**Comment by azzvx** [comment]
vps的防火墙配置了没有？除了22端口，其他端口vps默认不开放的。
233boy的脚本说明里有写：
`使用国际大厂的 VPS，请自行在安全组 (防火墙) 开放端口和 UDP 协议 (如果你要使用含有 mKCP 的传输协议)`

**Comment by colintheunique** [comment]
> vps的防火墙配置了没有？除了22端口，其他端口vps默认不开放的。 233boy的脚本说明里有写： `使用国际大厂的 VPS，请自行在安全组 (防火墙) 开放端口和 UDP 协议 (如果你要使用含有 mKCP 的传输协议)`

解决了打开端口问题，为什么我现在安装V2RAY之后，查看状态还是未运行呢？
5.1和之前版本都试过了


[root@vultr ~]# v2ray status

 V2Ray 状态: 未在运行

[root@vultr ~]# v2ray start

 V2Ray 已启动

[root@vultr ~]# v2ray status

 V2Ray 状态: 未在运行


[root@vultr ~]# systemd status v2ray
-bash: systemd: command not found
[root@vultr ~]# systemctl status v2ray
● v2ray.service - V2Ray Service
   Loaded: loaded (/etc/systemd/system/v2ray.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/v2ray.service.d
           └─10-donot_touch_single_conf.conf
   Active: active (running) since Mon 2022-10-31 15:22:47 CST; 6min ago
     Docs: https://www.v2fly.org/
 Main PID: 2745 (v2ray)
   CGroup: /system.slice/v2ray.service
           └─2745 /usr/local/bin/v2ray run -config /usr/local/etc/v2ray/confi...

Oct 31 15:22:47 vultr.guest systemd[1]: Stopped V2Ray Service.
Oct 31 15:22:47 vultr.guest systemd[1]: Started V2Ray Service.
Oct 31 15:22:47 vultr.guest v2ray[2745]: V2Ray 5.1.0 (V2Fly, a community-dri...)
Oct 31 15:22:47 vultr.guest v2ray[2745]: A unified platform for anti-censorship.
Oct 31 15:22:48 vultr.guest v2ray[2745]: 2022/10/31 15:22:48 [Warning] V2Ray...d
Oct 31 15:25:37 vultr.guest v2ray[2745]: 2022/10/31 15:25:37 *****:...d
Oct 31 15:27:10 vultr.guest v2ray[2745]: 2022/10/31 15:27:10 *****...d
Hint: Some lines were ellipsized, use -l to show in full.
[root@vultr ~]#  ss -tunlp| grep v2ray
tcp    LISTEN     0      128    [::]:20088              [::]:*                   users:(("v2ray",pid=2745,fd=3))
[root@vultr ~]# v2ray log

 按 Ctrl + C 即可退出...



这是安装V2ray的问题造成的吗？

**Comment by colintheunique** [comment]
> vps的防火墙配置了没有？除了22端口，其他端口vps默认不开放的。 233boy的脚本说明里有写： `使用国际大厂的 VPS，请自行在安全组 (防火墙) 开放端口和 UDP 协议 (如果你要使用含有 mKCP 的传输协议)`

跟安装时的mirror有关系吗？

**Comment by azzvx** [comment]
> systemctl status v2ray

用`systemctl status v2ray -l`看看完整日志，另外你IP暴露了

**Comment by colintheunique** [comment]
> > systemctl status v2ray
> 
> 用`systemctl status v2ray -l`看看完整日志，另外你IP暴露了

谢谢 orz
[root@vultr ~]# systemctl status v2ray -l
● v2ray.service - V2Ray Service
   Loaded: loaded (/etc/systemd/system/v2ray.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/v2ray.service.d
           └─10-donot_touch_single_conf.conf
   Active: active (running) since Mon 2022-10-31 15:52:46 CST; 2min 30s ago
     Docs: https://www.v2fly.org/
 Main PID: 29716 (v2ray)
   CGroup: /system.slice/v2ray.service
           └─29716 /usr/local/bin/v2ray run -config /usr/local/etc/v2ray/config.json

Oct 31 15:52:46 vultr.guest systemd[1]: Stopped V2Ray Service.
Oct 31 15:52:46 vultr.guest systemd[1]: Started V2Ray Service.
Oct 31 15:52:46 vultr.guest v2ray[29716]: V2Ray 5.1.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.19 linux/amd64)
Oct 31 15:52:46 vultr.guest v2ray[29716]: A unified platform for anti-censorship.
Oct 31 15:52:46 vultr.guest v2ray[29716]: 2022/10/31 15:52:46 [Warning] V2Ray 5.1.0 started
[root@vultr ~]#

**Comment by colintheunique** [comment]
> > systemctl status v2ray
> 
> 用`systemctl status v2ray -l`看看完整日志，另外你IP暴露了

另外我看网上有人说5.1版本有问题？

**Comment by azzvx** [comment]
防火墙配置呢？`firewall-cmd --list-all`
```
[root@racknerd-1df272 ~]# firewall-cmd --list-all
public (active)
  target: default
  icmp-block-inversion: no
  interfaces: eth0
  sources:
  services: dhcpv6-client ssh
  ports: 443/tcp 443/udp 5201/tcp 5201/udp 80/tcp 80/udp
  protocols:
  masquerade: no
  forward-ports:
  source-ports:
  icmp-blocks:
  rich rules:
```

**Comment by colintheunique** [comment]
> 防火墙配置呢？`firewall-cmd --list-all`
> 
> ```
> [root@racknerd-1df272 ~]# firewall-cmd --list-all
> public (active)
>   target: default
>   icmp-block-inversion: no
>   interfaces: eth0
>   sources:
>   services: dhcpv6-client ssh
>   ports: 443/tcp 443/udp 5201/tcp 5201/udp 80/tcp 80/udp
>   protocols:
>   masquerade: no
>   forward-ports:
>   source-ports:
>   icmp-blocks:
>   rich rules:
> ```

[root@vultr ~]# firewall-cmd --list-all
public
  target: default
  icmp-block-inversion: no
  interfaces:
  sources:
  services: dhcpv6-client ssh
  ports: 20088/tcp
  protocols:
  masquerade: no
  forward-ports:
  source-ports:
  icmp-blocks:
  rich rules:

[root@vultr ~]#
 这样

**Comment by azzvx** [comment]
你的防火墙激活了么？我看到你的结果`public`后面没有`(active)`字样。
```
[root@racknerd-1df272 ~]# firewall-cmd --state
running
[root@racknerd-1df272 ~]# firewall-cmd --get-active-zones
public
  interfaces: eth0
```

**Comment by colintheunique** [comment]
> 你的防火墙激活了么？我看到你的结果`public`后面没有`(active)`字样。
> 
> ```
> [root@racknerd-1df272 ~]# firewall-cmd --state
> running
> [root@racknerd-1df272 ~]# firewall-cmd --get-active-zones
> public
>   interfaces: eth0
> ```

[root@vultr ~]# firewall-cmd --list-all
public
target: default
icmp-block-inversion: no
interfaces:
sources:
services: dhcpv6-client ssh
ports: 20088/tcp
protocols:
masquerade: no
forward-ports:
source-ports:
icmp-blocks:
rich rules:

[root@vultr ~]# firewall-cmd --reload
success
[root@vultr ~]# firewall-cmd --list-all
public
target: default
icmp-block-inversion: no
interfaces:
sources:
services: dhcpv6-client ssh
ports: 20088/tcp
protocols:
masquerade: no
forward-ports:
source-ports:
icmp-blocks:
rich rules:

[root@vultr ~]# firewall-cmd --state;
running

[root@vultr ~]# systemctl status firewalld
● firewalld.service - firewalld - dynamic firewall daemon
Loaded: loaded (/usr/lib/systemd/system/firewalld.service; enabled; vendor preset: enabled)
Active: active (running) since Mon 2022-10-31 15:15:42 CST; 1h 35min ago
Docs: man:firewalld(1)
Main PID: 2292 (firewalld)
CGroup: /system.slice/firewalld.service
└─2292 /usr/bin/python2 -Es /usr/sbin/firewalld --nofork --nopid

Oct 31 15:15:42 vultr.guest systemd[1]: Starting firewalld - dynamic firewal....
Oct 31 15:15:42 vultr.guest systemd[1]: Started firewalld - dynamic firewall....
Oct 31 15:15:42 vultr.guest firewalld[2292]: WARNING: AllowZoneDrifting is e....
Oct 31 16:49:18 vultr.guest firewalld[2292]: WARNING: AllowZoneDrifting is e....
Hint: Some lines were ellipsized, use -l to show in full.
[root@vultr ~]#

**Comment by colintheunique** [comment]
> 你的防火墙激活了么？我看到你的结果`public`后面没有`(active)`字样。
> 
> ```
> [root@racknerd-1df272 ~]# firewall-cmd --state
> running
> [root@racknerd-1df272 ~]# firewall-cmd --get-active-zones
> public
>   interfaces: eth0
> ```

应该是激活了的，我也不知奥为啥没有active

firewall-cmd --state;查是active的状态

**Comment by azzvx** [comment]
`firewall-cmd --get-active-zones`的结果呢？

**Comment by colintheunique** [comment]
> firewall-cmd --get-active-zones

[root@vultr ~]# firewall-cmd --get-active-zones
[root@vultr ~]#


无返回

**Comment by azzvx** [comment]
你的防火墙没有应用到接口。应用到接口后看看结果。
```
firewall-cmd --permanent --zone=public --add-interface=eth0
firewall-cmd --reload
firewall-cmd --list-all
```

**Comment by azzvx** [comment]
或者换个版本看看。

**Comment by azzvx** [comment]
用这个可以测试VPS端口开了没有：https://www.yougetsignal.com/tools/open-ports/

**Comment by colintheunique** [comment]
> 用这个可以测试VPS端口开了没有：https://www.yougetsignal.com/tools/open-ports/

这个测20088接口是OPEN的

**Comment by FreeSlaver** [comment]
真的没撒必要折腾，都是一路折腾过来的人，从ss到v2ray到trojan，
从nginx反代，套域名，套CDN，套cloudflare，套各种协议，都没撒用。
还是用别人的机场把，便宜稳定，撒事都不用操心，<a href="https://user.by.ltd/aff.php?aff=17672">bywave</a>。
你那折腾的几个小时玩会游戏，刷会视频，学习下，它不香吗？
这个机场我18年末用到现在，当初有个套餐是80块钱250gb流量，包终生。
后面遇到3次还是几次故障，只要时间够久都会赔偿，有一次好像是赔偿了100多g。
我就白嫖到了现在，还有70g流量。当时冲了100块，现在还有35块余额了。
专业的事情还是交给专业的人干把。
